abi/erc20.go

```diff
@@ -1,5 +1,56 @@
-package abis
+package contracts
 
+import (
+	"errors"
+	"math/big"
+	"strconv"
+)
+
+type Erc20Params struct {
+	Symbol      string
+	TokenName   string
+	Decimals    int
+	TotalSupply *big.Int
+}
+
+func ParseERC20Params(args []string) (*Erc20Params, error) {
+	if len(args) != 4 {
+		return nil, errors.New("wrong amount of the parameters, the following parameters are mandatory SYMBOL TOKEN_NAME DECIMALS TOTAL_SUPPLY")
+	}
+	decimals, err := strconv.Atoi(args[2])
+	if err != nil {
+		return nil, err
+	}
+	totalSupply, ok := new(big.Int).SetString(args[3], 10)
+	if !ok {
+		return nil, errors.New("Cannot parse total supply")
+	}
+	totalSupply.Mul(totalSupply, new(big.Int).Exp(big.NewInt(10), big.NewInt(int64(decimals)), nil))
+	params := Erc20Params{
+		Symbol:      args[0],
+		TokenName:   args[1],
+		Decimals:    decimals,
+		TotalSupply: totalSupply,
+	}
+	return &params, nil
+}
+
+const ERC20_TEMPLATE = `pragma solidity ^0.5.2;
```

Comment:
Should this have a more specific name, since it's only one possible type of ERC20?